### PR TITLE
add smoke test in release workflow

### PR DESCRIPTION
**Pull request by wangzlei**
Automate smoke test in release workflow. Next can automate create & merge PR with new released layer ARNs.

### CODE REVIEW

File: sample-apps/java-wrapper-aws-sdk-terraform/layer.tf
Comment (anuraaga) [inline sample-apps/java-wrapper-aws-sdk-terraform/layer.tf:1]: Can you run `terraform fmt --recursive`?

File: .github/workflows/release.yml
Comment (anuraaga) [inline .github/workflows/release.yml:268]: `git commit` and `push` at the end?

Comment (anuraaga) [reply .github/workflows/release.yml:268]: Well I guess we'd want one commit, not one per app though

Comment (anuraaga) [reply .github/workflows/release.yml:268]: Talked offline, let's follow up for this

Comment (wangzlei) [reply .github/workflows/release.yml:268]: we can use github action https://github.com/marketplace/actions/create-pull-request

But need to maintain a map from layer_kind to the path of layer.tf.

File: .github/workflows/release.yml
Comment (wangzlei) [inline .github/workflows/release.yml:128]: actually we only need run smoke test for one specific layer kind, any good idea to skip the others?

File: .github/workflows/release.yml
Comment (willarmiros) [inline .github/workflows/release.yml:189]: I'll leave it up to you, but maybe it would be cleaner if you could refactor this to have 1 step that always caches the go modules (regardless of `matrix.language`), and then these conditional caches would only apply to that language's artifacts.

**Review by willarmiros — APPROVED** [review]
Just one comment, but won't block on it.

**Review by anuraaga — APPROVED** [review]

**Pull request merged**
